crates/uv/src/commands/publish.rs

```diff
@@ -75,6 +75,10 @@ pub(crate) async fn publish(
         (username, password)
     };
 
+    if password.is_some() && username.is_none() {
+        bail!("You need to provide a username with a password, use `--token` for tokens");
```

Comment:
I think we should suggest `--user` and `--token`, and note the env vars since it's common in GHA:

> Attempted to publish with a password, but no username. Either provide a username with `--user` (`UV_PUBLISH_USERNAME`), or use `--token` (`UV_PUBLISH_TOKEN`) in lieu of a password.

Reply:
This case only appears when you use only a password and not a username, despite token vs username/password are clearly documented in the pypi docs and our docs, and other indexes that i have tested providing clear guidance on username and password values. This is a fringe case and I'd rather spend our efforts on more impactful UX aspects, such as showing good messages after 403 errors, guiding users to the correct github actions trusted publishing settings, skip-existing/retries, etc.